resolwe_bio/processes/reads_processing/alignmentsieve.py

```diff
@@ -0,0 +1,151 @@
+"""Filter BAM files using alignmentSieve from deeptools"""
+import os
+from pathlib import Path
+
+from plumbum import TEE
+
+from resolwe.process import (
+    Process,
+    SchedulingClass,
+    Persistence,
+    Cmd,
+    StringField,
+    IntegerField,
+    FileField,
+    DataField,
+)
+
+
+class AlignmentSieve(Process):
+    """Filter alignments of BAM files according to specified parameters
+
+    Optionally, it can output a BEDPE file. Program is bundled with
+    deeptools. See
+    https://deeptools.readthedocs.io/en/develop/content/tools/alignmentSieve.html
+    for more details.
+    """
+
+    slug = "alignmentsieve"
+    name = "alignmentSieve"
+    process_type = "data:alignment:bam:"
+    version = "1.0.0"
+    category = "Read Processing"
+    data_name = 'Sieved BAM ({{ alignment|sample_name|default("?") }})'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+    entity = {
+        "type": "sample",
+    }
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {"docker": {"image": "resolwebio/common:1.6.0"}},
+        "resources": {
+            "cores": 10,
+            "memory": 16384,
+            "network": True,
+        },
+    }
+
+    class Input:
+        """Input fields to process AlignmentSieve."""
+
+        alignment = DataField("alignment:bam", label="Alignment BAM file")
+        min_fragment_length = IntegerField(
+            label="--minFragmentLength",
+            description="The minimum fragment length needed for "
+            "read/pair inclusion. This option is primarily useful in "
+            "ATACseq experiments, for filtering mono- or di-nucleosome "
+            "fragments. (Default: 0)",
+            default=0,
+        )
+        max_fragment_length = IntegerField(
```

Comment:
What happens if `max_fragment_length` is set to a smaller value than `min_fragment_length`?

Reply:
Not sure, but I think you get more or less all the reads.